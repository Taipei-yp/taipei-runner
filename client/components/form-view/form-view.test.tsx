import React from "react";
import TestRenderer, { act } from "react-test-renderer";
import { FormView } from ".";

describe("Component FormView", () => {
  const onSubmit = jest.fn();
  const formView = TestRenderer.create(
    <FormView fields={[]} onSubmit={onSubmit} />,
  );

  it("correct render", () => {
    expect(formView.toJSON()).toMatchSnapshot();
  });

  it("onsubmit work", () => {
    act(formView.root.props.onSubmit);
    expect(onSubmit).toBeCalledTimes(1);
  });
});
