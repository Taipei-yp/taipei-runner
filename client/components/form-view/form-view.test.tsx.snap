// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component FormView correct render 1`] = `
<div
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
>
  <form
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    onSubmit={[Function]}
  >
    <button
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
